test/integration-no-mock/config-management/env-config-create.js

```diff
@@ -0,0 +1,404 @@
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+const moment = require('moment');
+const path = require('path');
+
+const ApiService = require('./../../ApiService');
+const ConfigManagementHelper = require('./../../ConfigManagementHelper');
+
+const EnvHelper = ConfigManagementHelper.env;
+const AppHelper = ConfigManagementHelper.app;
+const { execCmdWoMocks, randomStrings } = require('./../../TestsHelper');
+
+module.exports = () => {
+  const appName = randomStrings.appName();
+  const serviceIds = [];
+
+  before('setup app', (done) => {
+    AppHelper.createInTestsOrg({ name: appName }, done);
+  });
+
+  after('remove app and services', (done) => {
+    async.series([
+      (next) => {
+        execCmdWoMocks(`app delete ${appName} --no-prompt`, null, next);
+      },
+      (next) => {
+        async.each(
+          serviceIds,
+          (serviceId, next) => {
+            ApiService.services.remove(serviceId, next);
+          },
+          next
+        );
+      }
+    ], done);
+  });
+
+  it('settings only should succeed', (done) => {
+    const envSettings = EnvHelper.buildSettings();
+    const env = {
+      schemaVersion: '1.0.0',
+      configType: 'environment',
+      settings: envSettings
+    };
+
+    const envName = randomStrings.envName();
+    let envId;
+
+    async.series([
+      (next) => {
+        EnvHelper.createFromConfig(envName, env, appName, (err, id) => {
+          if (err) {
+            return next(err);
+          }
+
+          envId = id;
+          next();
+        });
+      },
+      (next) => {
+        EnvHelper.assertEnvOnly(env, envId, envName, next);
+      }
+    ], done);
+  });
+
+  it('settings and internal collections wo system collections should succeed', (done) => {
+    const collList = EnvHelper.buildValidInternalCollectionsList(3, false);
+    const envSettings = EnvHelper.buildSettings();
+    const env = {
+      schemaVersion: '1.0.0',
+      configType: 'environment',
+      settings: envSettings,
+      collections: ConfigManagementHelper.common.buildConfigEntityFromList(collList)
+    };
+
+    const envName = randomStrings.envName();
+    let envId;
+
+    async.series([
+      (next) => {
+        EnvHelper.createFromConfig(envName, env, appName, (err, id) => {
+          if (err) {
+            return next(err);
+          }
+
+          envId = id;
+          next();
+        });
+      },
+      (next) => {
+        EnvHelper.assertEnvOnly(env, envId, envName, next);
+      },
+      (next) => {
+        EnvHelper.assertCollections(envId, collList, collList.length + 2, null, next);
+      }
+    ], done);
+  });
+
+  it('internal collections only plus system collections should succeed', (done) => {
```

Comment:
If the purpose of the test if to verify that user and _blob collections will be updated, we should pass a different type of permission for them - e.g. 'private'. Otherwise, currently the test will pass even if _blob and user collections are not updated from the cli.

Reply:
It is more to ensure that all internal collections will be created correctly, that error won't occur because of system collections ('cause unlike other collections they should be updated, not created) and that there will be no errors just because we have only collections (not a full config). It's just a base case, really.